app/controllers/reservations_controller.rb

```diff
@@ -24,8 +24,8 @@ def update
 
   def parse_json_request
     @request_data = JSON.parse(request.body.read)
-  rescue JSON::ParserError
-    head :bad_request
+  rescue JSON::ParserError => e
+    render :json => { errors: { request_body: ["invalid JSON: #{e.message}"] } }, status: :bad_request
```

Comment:
One small suggestion: I had to read this line through a couple of times to be sure what it was doing, and what was being passed where. Might it be worth pulling out, say, a method that takes the exception as a parameter and returns the response hash? For example:

```ruby
def parse_json_request
  @request_data = JSON.parse(request.body.read)
rescue JSON::ParserError
  head :bad_request
rescue JSON::ParserError => e
  render :json => parser_error_response(e), status: :bad_request
end

def parser_error_response(error)
  { errors: { request_body: ["Invalid JSON: #{e.message}"] } }
end